sherpa/astro/io/pyfits_backend.py

```diff
@@ -1465,3 +1465,101 @@ def set_arrays(filename, args, fields=None, ascii=True, clobber=False):
     hdu = fits.table_to_hdu(tbl)
     hdu.name = 'TABLE'
     hdu.writeto(filename, overwrite=True)
+
+
+def _add_header(hdu, header):
+    """Add the header items to the HDU.
+
+    Parameters
+    ----------
+    hdu : astropy HDU
+    header : list of sherpa.astro.io.xstable.HeaderItem
+    """
+
+    for hdr in header:
+        card = [hdr.name, hdr.value]
+        if hdr.desc is not None or hdr.unit is not None:
+            comment = "" if hdr.unit is None else f"[{hdr.unit}] "
+            if hdr.desc is not None:
+                comment += hdr.desc
+            card.append(comment)
+
+        hdu.header.append(tuple(card))
+
+
+def _create_primary_hdu(hdu):
+    """Create a PRIMARY HDU.
+
+    Parameters
+    ----------
+    hdu : sherpa.astro.io.xstable.TableHDU
+       Any data is ignored.
+
+    Returns
+    -------
+    out : astropy.io.fits.PrimaryHDU
+
+    """
+
+    out = fits.PrimaryHDU()
+    _add_header(out, hdu.header)
+    return out
+
+
+def _create_table_hdu(hdu):
```

Comment:
I beleive that this might not be the easiest way to write the code, but it works and gives correct answers (accordig to the tests) and speed is not a concern, so I'm not going to nitpick if e.g. units are passed to the initial table or are added "by hand" to the header later.